### ISSUE TITLE
Show changelog only for changes in a subfolder

### ISSUE THREAD

**Issue by davidsneighbour** [absolute-version/commit-and-tag-version#149]
I have a mono repo which sets up various modules in subfolders:

/modules/module1
/modules/module2
...

With tags formatted as `modules/module1/v1.2.3` and `modules/module2/v1.2.3`. 

Is it possible to enable changelog generation that loads only those changes within one of these folders between tags with that name? I am able to retrieve the current tag and the next tag, so that would probably just be done with CLI parameters, but I am unsure about the changelog generation. Is there a way to "pipe" a custom list of commits into the script?

**Comment by TimothyJones** [comment]
I don’t believe there currently is an easy way, but I agree, this would be a clearly useful use case

**Comment by gurukulkarni** [comment]
Exactly, I think there are generally much less tools supporting monorepos this way and makes me think either the tooling has not caught up or I am incapable of finding it.
What would be amazing is, not only restrict to subfolders but also only consider commits that change files in given subfolders and only take tags pointing to commits that change said subfolder.

This would allow something like : 
```
repo >
   module-1 >
   		m1-1
   		m1-2 
   module-2 >
         m2-1
         m2-2
```
then when run against module-1 just consider commits and tags from that one and bump and create changelog restricting to only that directory.
I'm not good with javascript but will try to have a look when I have time if this is possible to do :)

**Comment by mattpfeffer** [comment]
+1 for some kind of support in this area. In my monorepo I tend to use the package name as the commit scope. So as a workaround I use the following script on the 'postchangelog' lifecycle hook to filter the changelog based on scope:

````js
import fs from 'fs';
import readline from 'readline';

const scope = process.cwd().match(/([^/]*)$/, '')[0];

const changelog = fs.createReadStream(`./CHANGELOG.md`);

const reader = readline.createInterface({
    input: changelog,
    crlfDelay: Infinity
});

let filtered = '';

for await (const line of reader) {
    if (line.startsWith('* **')) {
        const pattern = `**${scope}:** `;
        if (line.includes(pattern)) filtered += line.replace(pattern, '') + '\n';
    } else {
        filtered += line + '\n';
    }
}

fs.writeFileSync(`./CHANGELOG.md`, filtered);
````
This assumes that the scope name, sub package name and folder are all equal. There are some obvious limitations to this approach but for the most part it works.

**Comment by CaiWenlie** [comment]
try `npx lerna version --conventional-commits`

**Comment by rix0rrr** [comment]
There is the `--path` argument. Does that do what you need?

https://github.com/absolute-version/commit-and-tag-version/blob/400e3c17616cfc2481c5a17dad85c0d4d67a49f1/lib/lifecycles/bump.js#L194
